tests/test_app/tests/test_api.py

```diff
@@ -244,22 +246,50 @@ def test_reverting_test_model(self):
-        self.assertEqual(response.data[1]['revision']['user']['username'], 'user2')
-        self.assertIsNotNone(response.data[1]['revision']['comment'])
-        self.assertEqual(response.data[1]['field_dict']['name'], 'Foo 1.2.1')
-        self.assertIsNotNone(response.data[2]['id'])
-        self.assertIsNotNone(response.data[2]['revision']['date_created'])
-        self.assertEqual(response.data[2]['revision']['user']['id'], 1)
-        self.assertEqual(response.data[2]['revision']['user']['username'], 'user1')
-        self.assertIsNotNone(response.data[2]['revision']['comment'])
-        self.assertEqual(response.data[2]['field_dict']['name'], 'Foo 1.2.0')
+        self.assertEqual(response.data["count"], 3)
+        self.assertIsNotNone(response.data["results"][0]['id'])
+        self.assertIsNotNone(response.data["results"][0]['revision']['date_created'])
+        self.assertEqual(response.data["results"][0]['revision']['user']['id'], 2)
+        self.assertEqual(response.data["results"][0]['revision']['user']['username'], 'user2')
+        self.assertIsNotNone(response.data["results"][0]['revision']['comment'])
+        self.assertEqual(response.data["results"][0]['field_dict']['name'], 'Foo 1.2.0')
+        self.assertIsNotNone(response.data["results"][1]['id'])
+        self.assertIsNotNone(response.data["results"][1]['revision']['date_created'])
+        self.assertEqual(response.data["results"][1]['revision']['user']['id'], 2)
+        self.assertEqual(response.data["results"][1]['revision']['user']['username'], 'user2')
+        self.assertIsNotNone(response.data["results"][1]['revision']['comment'])
+        self.assertEqual(response.data["results"][1]['field_dict']['name'], 'Foo 1.2.1')
+        self.assertIsNotNone(response.data["results"][2]['id'])
+        self.assertIsNotNone(response.data["results"][2]['revision']['date_created'])
+        self.assertEqual(response.data["results"][2]['revision']['user']['id'], 1)
+        self.assertEqual(response.data["results"][2]['revision']['user']['username'], 'user1')
+        self.assertIsNotNone(response.data["results"][2]['revision']['comment'])
+        self.assertEqual(response.data["results"][2]['field_dict']['name'], 'Foo 1.2.0')
+
+    def test_pagination(self):
+        """
+        Ensure that the pagination works as expected, and that individual revisions can be obtained.
+        """
+        create_url = reverse('testmodelcustom-list')
+
+        response = self.client.post(create_url, {'name': 'Foo 1.2.0'}, format='json')
+        pk = response.data['id']
+
+        update_url = reverse('testmodelcustom-detail', kwargs={'pk': response.data['id']})
+
+        for letter in string.ascii_lowercase:
+            self.client.patch(update_url, {'name': letter}, format='json')
+
+        for page, count in [(1, 10), (2, 10), (3, 7)]:
+            history_url = f"http://testserver/test-app/test-models-custom/{pk}/history/?page={page}"
```

Comment:
i could not get `reverse` to work for the nested route

Reply:
Fixed with

```python
history_base_url = reverse('testmodelpaginated-history', kwargs={'pk': pk})
for page, count in [(1, 10), (2, 10), (3, 7)]:
    query_kwargs = {'page': page}
    history_url = f'{history_base_url}?{urlencode(query_kwargs)}'
```